R/github_token.R

```diff
@@ -296,3 +296,52 @@ scold_for_scopes <- function(scopes) {
   )
   ui_bullets(message)
 }
+
+#' Lock and unlock a branch on GitHub
+#'
+#' @description
+#' These functions lock and unlock a branch on GitHub so that it's not possible
+#' for anyone to make any changes. This is used as part of the release process
+#' to ensure that you don't accidentally merge any pull requests or push any
+#' commits while you are waiting for CRAN to get back to you.
+#'
+#' You must be an admin or an owner of the repo in order to lock/unlock
+#' a branch.
+#'
+#' @export
+#' @param branch The branch to lock/unlock. If not supplied, uses the
+#'   default branch which is usually "main" or "master".
+gh_lock_branch <- function(branch = NULL) {
+  cfg <- github_remote_config(github_get = TRUE)
+  repo <- target_repo(cfg)
+  branch <- branch %||% git_default_branch_(cfg)
+
+  invisible(gh::gh(
+    "PUT /repos/{owner}/{repo}/branches/{branch}/protection",
+    owner = repo$repo_owner,
+    repo = repo$repo_name,
```

Comment:
We usually do a different move to create these `gh::gh()` calls for the target repo, but I am happy to do this consistency pass myself before we merge.

Reply:
Thanks!